pkg/iac/terraform/state/terraform_state_reader.go

```diff
@@ -41,6 +41,7 @@ type TerraformStateReader struct {
 	progress       output.Progress
 	filter         filter.Filter
 	alerter        *alerter.Alerter
+	keys           []string
```

Comment:
I think we should find a better naming for that, `keys` sounds too generic for me.

Since we only use theses "keys" to return the length, why not explicitly storing the number of states here and returning it directly in `SourceCount` ? In `retrieveForState` you'll only have to increment a counter.

So we can rename the unexported `keys` attribute to `stateCount` WDYT ?